hermes-common/src/main/java/pl/allegro/tech/hermes/infrastructure/zookeeper/cache/HierarchicalCacheLevel.java

```diff
@@ -94,12 +112,51 @@ private void addSubcache(String path, String cacheName) throws Exception {
         } finally {
             writeLock.unlock();
         }
+    }
 
+    private void printOrphanedChildren(String path) {
+        try {
+            List<String> children = curatorClient.getChildren().forPath(path);
+            logger.warn("Nodes with empty parent {}: {}",
+                    path, children.stream().map(Object::toString).collect(Collectors.joining(",")));
+            printChildrenWithEmptyParentRecursively(path, children);
+        } catch (KeeperException.NoNodeException e) {
+            // probably already removed
+        } catch (Exception e) {
+            logger.warn("Could not receive list of children for path {} due to error", path, e);
+        }
+    }
+
+    private void printChildrenWithEmptyParentRecursively(String pathWithEmptyParent, List<String> children) {
+        children.forEach( c -> {
+            try {
+                String nextPath = pathWithEmptyParent + "/" + c;
+                logger.warn("Node with empty parent: {}", nextPath);
+                List<String> nextChildren = curatorClient.getChildren().forPath(nextPath);
+
+                printChildrenWithEmptyParentRecursively(nextPath, nextChildren);
+            } catch (KeeperException.NoNodeException e) {
+                // probably already removed
```

Comment:
"probably" tells that you are not 100% sure :)
IMO the exception should be logged.

Reply:
Ok, I’ve added log here.